proxmin/algorithms.py

```diff
@@ -8,15 +8,15 @@
 logging.basicConfig()
 logger = logging.getLogger("proxmin")
 
-def pgm(X0, prox_f, step_f, accelerated=False, relax=None, e_rel=1e-6, max_iter=1000, traceback=None):
+def pgm(X, prox_f, step_f, accelerated=False, relax=None, e_rel=1e-6, max_iter=1000, traceback=None):
     """Proximal Gradient Method
 
     Adapted from Combettes 2009, Algorithm 3.4.
     The accelerated version is Algorithm 3.6 with modifications
     from Xu & Yin (2015).
 
     Args:
-        X0: initial X
+        X: initial X, will be updated
```

Comment:
With this change, line 58 below should be:
```
X[:] = prox_f(_X, step_f)
```

Reply:
Indeed. That works often because `prox_f` internally overwrites `X`, but this version works regardless.